### PR TITLE
Deemphasize processors and syntax option in the docs

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Ref: https://github.com/stylelint/stylelint/issues/3587

> Is there anything in the PR that needs further explanation?

This PR deemphasizes processors and syntax option in the docs (and emphasizes stylelint's built-in support for inferring syntax and extracting styles). It also continues to roll-out the simplified writing styled we started using on the homepage.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,30 +27,31 @@ It's mighty because it:
 
 It's easy to get started.
 
-First, decide how you want to use stylelint:
+First, decide if you want to use stylelint:
```
Comment (ntwb) [inline README.md]: I think the user has _already_ decided to use stylelint if they are at this point...

The following 5 bullet items are "how" to use stylelint, they are not sentences, statements, or quotes to aid a user in deciding "if" they should use stylelint.

I suggest reverting this change back to `how`

File: README.md [hunk 2]
```diff
@@ -27,30 +27,31 @@ It's mighty because it:
...
 
 -   [on the command line](docs/user-guide/cli.md)
 -   [in your text editor](docs/user-guide/complementary-tools.md#editor-plugins), for example in VS Code
--   [in for your build tool](docs/user-guide/complementary-tools.md#build-tool-plugins), for example in webpack
+-   [in for your build tool](docs/user-guide/complementary-tools.md#build-tool-plugins), for example in Webpack
```
Comment (ntwb) [inline README.md]: `webpack` is like stylelint, all spelled with lowercase `webpack`

See https://webpack.js.org/branding/#the-name

File: README.md (hunk 1 above)
Comment (jeddy3) [reply README.md]: Yep, it's a odd one. I was following [this guidance](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#bullet-points-steps):

> - you always use a lead-in line
> - the bullets make sense running on from the lead-in line

"First, decide if you want to use stylelint on the CLI"
"First, decide if you want to use stylelint in your build tool"

Whereas "how" ends up meaning something else "First, decide how you want to use stylelint on the CLI". 

There's probably a better way of writing the lead-in line to remove the oddity you mentioned, but I can't think of one right now :) So, I'll revert the change.

Comment (ntwb) [reply README.md]: Ha, I guess so, language is hard and I only speak one!

File: README.md [hunk 3]
```diff
@@ -10,10 +10,10 @@ It's mighty because it:
 
 -   has over **160 built-in rules** to catch errors, apply limits and enforce stylistic conventions
 -   understands the **latest CSS syntax** including custom properties and level 4 selectors
--   parses **CSS-like syntaxes** like SCSS, Sass, Less and SugarSS
 -   extracts **embedded styles** from HTML, markdown and CSS-in-JS object & template literals
--   automatically **fixes** some violations (*experimental feature*)
+-   parses **CSS-like syntaxes** like SCSS, Sass, Less and SugarSS
 -   supports **plugins** so you can create your own rules or make use of plugins written by the community
+-   automatically **fixes** some violations (*experimental feature*)
```
[the remark this answers is not in the document]
Comment (hudochenkov) [reply README.md]: We have one important [issue](https://github.com/stylelint/stylelint/issues/2643) to solve, before we can recommend `--fix` without risk.

[the remark this answers is not in the document]
Comment (jeddy3) [reply README.md]: > We have one important issue to solve

Yep, definitely that one.

We probably need to resolve the whole order-of-rules-being-important thing too e.g. `indentation` needs to run after a lot of other rules.

**Pull request merged**